Sources/ProjectDescription/Entitlements.swift

```diff
@@ -9,6 +9,9 @@ public enum Entitlements: Codable, Equatable {
     /// A dictionary with the entitlements content. Tuist generates the .entitlements file at the generation time.
     case dictionary([String: Plist.Value])
 
+    /// A user defined xcconfig variable map to .entitlements file
+    case xcconfig(String? = nil)
```

Comment:
minor perhaps we can call it variable as this can be inherited from target settings, project settings, or Configs

```suggestion
    case variable(String)
```

Curious why it's being optional? 

- `entitlements` is already an optional property on `Target`
- If it isn't specified it would result in not setting the build setting `CODE_SIGN_ENTITLEMENTS` which would be the equivalent of `$(inherited)` or `$(CODE_SIGN_ENTITLEMENTS)`

Reply:
My idea from having this an option is set it to a default value `${CODE_SIGN_ENTITLEMENTS}`,

You're right, on target entitlement is already optional, but for appclips the linter make it required.

I'll add a commit that renames the `xcconfig` to `variable` and remove the optional to make that variable name set explicit instead of implicit.